Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift

```diff
@@ -227,12 +227,26 @@ internal final class CommentRepliesDataSource: ValueCellDataSource {
     return nil
   }
 
+  /**
+   Returns `true` when  the `replies`section of the data source is empty.
+
+   - parameter in: `UITableView` object that we need the number of sections from.
+
+   - returns: A  `Bool` which is only true if the number of items in the `replies` section is 0.
+   */
+  public func isRepliesSectionEmpty(in tableView: UITableView) -> Bool {
+    if self.numberOfSections(in: tableView) > Section.replies.rawValue {
+      return self.numberOfItems(in: Section.replies.rawValue) == 0
+    }
+    return true
```

Comment:
Ah so this method was written because our data source is unfortunately a bit inflexible. Checking the number of items in an empty section will result in a fatal error (index out of bounds). That's why I have to perform that check for the number of sections in the table view being greater than the index of the replies section. It's essentially a hacky way of verifying the section exists and whether there are any items in there after.

On a similar note, I have to pass the tableview in here because there is no way for me to check the current number of sections in the data source without using a method that uses that parameter.